### PR TITLE
WIP - Launch

### PR DESCRIPTION
**Pull request by edewit**
New endpoint that "launches" osio project, need some more tests

### CODE REVIEW

**Comment by gastaldi** [comment]
Can you merge these changes in the osio-missioncontrol branch? See https://github.com/fabric8-launcher/launcher-backend/pull/70

File: addons/new-osio-addon/pom.xml [hunk 1]
```diff
@@ -40,5 +45,20 @@
...
     </dependency>
+    <dependency>
+      <groupId>io.fabric8.launcher</groupId>
+      <artifactId>launcher-core-impl</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>io.fabric8.launcher</groupId>
+      <artifactId>launcher-service-openshift-impl</artifactId>
+      <scope>test</scope>
```
Comment (gastaldi) [inline addons/new-osio-addon/pom.xml]: Aren't these services transitive dependencies in core-impl?

File: addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/providers/KubernetesClientProvider.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import io.fabric8.launcher.osio.EnvironmentVariables;
+
+import static io.fabric8.launcher.base.identity.IdentityFactory.createFromToken;
+import static io.fabric8.launcher.base.identity.TokenIdentity.removeBearerPrefix;
+
+@RequestScoped
+public class KubernetesClientProvider {
+
+    @Produces
+    public KubernetesClient createOpenshiftClient(HttpServletRequest request) {
```
Comment (gastaldi) [inline addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/providers/KubernetesClientProvider.java]: Not needed. You can use @Inject OpenshiftService and call getOpenshiftClient

File: addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/steps/GitSteps.java [hunk 3]
```diff
@@ -1,25 +1,103 @@
...
+            Path projectLocation = projectile.getProjectLocation();
+
+            // Add logged user in README.adoc
+            Path readmeAdocPath = projectLocation.resolve("README.adoc");
+            if (Files.exists(readmeAdocPath)) {
+                try {
+                    String content = new String(Files.readAllBytes(readmeAdocPath));
+                    Map<String, String> values = new HashMap<>();
+                    values.put("loggedUser", gitService.getLoggedUser().getLogin());
+                    String newContent = new StrSubstitutor(values).replace(content);
```
Comment (gastaldi) [inline addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/steps/GitSteps.java]: See the osio-missioncontrol branch. the lang3 dep is not needed

**Comment by edewit** [comment]
closed in favor of #84

File: addons/new-osio-addon/pom.xml (hunk 1 above)
Comment (edewit) [reply addons/new-osio-addon/pom.xml]: only git is, changed in #84

File: addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/providers/KubernetesClientProvider.java (hunk 2 above)
Comment (edewit) [reply addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/providers/KubernetesClientProvider.java]: fixed in #84

File: addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/steps/GitSteps.java (hunk 3 above)
Comment (edewit) [reply addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/steps/GitSteps.java]: fixed in #84

**Pull request closed without merging**